source/renderer/app/components/widgets/InView.tsx

```diff
@@ -0,0 +1,34 @@
+import { useRef, useState, useCallback } from 'react';
+
+type Props = {
+  children: (props: {
+    isInViewport: boolean;
+    setTargetRef: (target: HTMLElement) => void;
+  }) => JSX.Element;
+};
+
+export const InView = ({ children }: Props) => {
```

Comment:
1. I can see that below you're using term `viewport` rather than `view`. Would it make sense to call the component `InViewport` for consistency?

2. Did you consider writing this as a custom hook instead of a component with render prop?